source/Ship.cpp

```diff
@@ -1302,7 +1302,7 @@ vector<string> Ship::FlightCheck() const
 			checks.emplace_back("afterburner only?");
 		if(!thrust && !afterburner)
 			checks.emplace_back("reverse only?");
-		if(!generation && !solar && !consuming)
+		if(energy <= battery)
```

Comment:
The posted ship has Hai shield regenerators with the "energy consumption" attribute, which is causing the `generation` variable to go negative. `!generation` is synonymous with `generation != 0`, when in reality we want it to be `generation <= 0` because of how `generation` can go negative.
```suggestion
		if(generation <= 0 && !solar && !consuming)
```

Reply:
we also need to check if solar and consuming are <= 0, not just == 0 (for plugin-ins, at least, even if the main game doesn't do that). At which point you have `energy <= battery`.

Reply:
I have been through the logic three times on three different days from at least two angkes of approach. I am convinced the proposed PR is correct and as minimal as it can be.

Reply:
e.g. if `solar` is non-zero, but not high enough to overcome the negativeness of `generation`, we still need to show the warning. Your proposed change would not do that.

Reply:
That checks out.